pandas/core/groupby/groupby.py

```diff
@@ -431,11 +431,52 @@ def __init__(self, groupby: GroupBy) -> None:
         self._groupby = groupby
 
     def __call__(self, *args, **kwargs):
-        def f(self):
-            return self.plot(*args, **kwargs)
+        # Special case for scatter plots to enable automatic colors in groupby context
+        if kwargs.get("kind") == "scatter":
```

Comment:
From the issue report, this plot works with `kine="line"`. Why do we need all this logic for `scatter` but it just work for `line`?

Reply:
Great! This is much nicer!